### PR TITLE
openssl-ts: add page

### PR DESCRIPTION
**Pull request by 0x1080**
- [x] The page (if new), does not already exist in the repo.
- [x] The page is in the correct platform directory (`common/`, `linux/`, etc.)
- [x] The page has 8 or fewer examples.
- [x] The PR title conforms to the recommended [templates](/tldr-pages/tldr/blob/main/CONTRIBUTING.md#commit-message).
- [x] The page follows the [content guidelines](/tldr-pages/tldr/blob/main/CONTRIBUTING.md#guidelines).
- [x] The page description includes a link to documentation or a homepage (if applicable).

### CODE REVIEW

**Comment by CleanMachine1** [comment]
There are going to be some changes here definitely, please stick with us as YOU apply the suggestions we all agree on.

File: pages/common/openssl-ts.md [hunk 1]
```diff
@@ -0,0 +1,20 @@
+# openssl ts
+
+> OpenSSL command to generate and verify timestamps.
+> More information: <https://www.openssl.org/docs/manmaster/man1/openssl-ts.html>.
+
+- Generate sha512 timestamp request of `file` (output to `file.tsq`):
```
Comment (CleanMachine1) [inline pages/common/openssl-ts.md:6]:
```suggestion
- Generate SHA-512 timestamp request of `file` (output to `file.tsq`):
```

File: pages/common/openssl-ts.md [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+`openssl ts -query -data {{file}}  -sha512  -out {{file.tsq}}`
```
Comment (CleanMachine1) [inline pages/common/openssl-ts.md:8]:
```suggestion
`openssl ts -query -data {{path/to/file}}  -sha512  -out {{path/to/file.tsq}}`
```

File: pages/common/openssl-ts.md [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+
+- Check timestamp response file `file.tsr`:
```
Comment (CleanMachine1) [inline pages/common/openssl-ts.md:10]: I don't like this sentence @bl-ue Any suggestions?

File: pages/common/openssl-ts.md [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+
+`openssl ts -reply -in {{file.tsr}} -text`
```
Comment (CleanMachine1) [inline pages/common/openssl-ts.md:12]:
```suggestion
`openssl ts -reply -in {{path/to/file.tsr}} -text`
```

File: pages/common/openssl-ts.md [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+
+- Verify timestamp request `file.tsq` and timestamp response `file.tsr` from server with ssl certificate `cert.pem`:
+
+`openssl ts -verify -in {{file.tsr}} -queryfile {{file.tsq}} -partial_chain -CAfile {{cert.pem}}`
```
Comment (CleanMachine1) [inline pages/common/openssl-ts.md:16]:
```suggestion
`openssl ts -verify -in {{path/to/file.tsr}} -queryfile {{path/to/file.tsq}} -partial_chain -CAfile {{path/to/cert.pem}}`
```

File: pages/common/openssl-ts.md [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+
+- Create time stamp response for request `file.tsq` using key `tsakey.pem` and signing certificate `tsacert.pem` (output to `file.tsr`):
+
+`openssl ts -reply -queryfile {{file.tsq}} -inkey {{tsakey.pem}} -signer tsacert.pem -out {{file.tsr}}`
```
Comment (CleanMachine1) [inline pages/common/openssl-ts.md:20]:
```suggestion
`openssl ts -reply -queryfile {{path/to/file.tsq}} -inkey {{path/to/tsakey.pem}} -signer tsacert.pem -out {{path/to/file.tsr}}`
```

**Review by CleanMachine1 — COMMENTED** [review]
Definitely more changes to me made

File: pages/common/openssl-ts.md (hunk 2 above)
Comment (0x1080) [reply pages/common/openssl-ts.md:8]: Why do some tldr pages (eg openssl, gpg) use simple filenames like {{public.gpg}}, whereas others use {{path/to/file}}? Is there a convention based on the context?

File: pages/common/openssl-ts.md (hunk 3 above)
Comment (0x1080) [reply pages/common/openssl-ts.md:10]: How about this?
- Check date and metadata of timestamp response file `file.tsr`:

File: pages/common/openssl-ts.md (hunk 2 above)
Comment (0x1080) [reply pages/common/openssl-ts.md:8]: Thank you so much for reviewing my pull request ^_^

File: pages/common/openssl-ts.md (hunk 3 above)
Comment (CleanMachine1) [reply pages/common/openssl-ts.md]: I like that you are inputing, my problem is the `file file` bit

Comment (0x1080) [reply pages/common/openssl-ts.md]: Lol, I completely missed that.

Comment (0x1080) [reply pages/common/openssl-ts.md]: - Check date and metadata from `file.tsr`, the timestamp response file:

How about this?

Comment (CleanMachine1) [reply pages/common/openssl-ts.md]: Well I haven't used this tool, so I don't understand the context but that doesn't read right to me

- Check the date and metadata from a specified timestamp response file

Comment (0x1080) [reply pages/common/openssl-ts.md]: Are you sure it's a good idea not to reference that the timestamp response file is the argument with the .tsr extension?

You submit the timestamp request file (.tsq) to the time stamp server, it returns a time stamp request file (.tsr). This shows the date on the tsr and the metadata, but it doesn't verify the server's signature.

Here is a sample output for the command:

Using configuration from /usr/lib/ssl/openssl.cnf
Status info:
Status: Granted.
Status description: unspecified
Failure info: unspecified

TST info:
Version: 1
Policy OID: tsa_policy1
Hash Algorithm: sha512
Message data:
    0000 - cf 83 e1 35 7e ef b8 bd-f1 54 28 50 d6 6d 80 07   ...5~....T(P.m..
    0010 - d6 20 e4 05 0b 57 15 dc-83 f4 a9 21 d3 6c e9 ce   . ...W.....!.l..
    0020 - 47 d0 d1 3c 5d 85 f2 b0-ff 83 18 d2 87 7e ec 2f   G..<]........~./
    0030 - 63 b9 31 bd 47 41 7a 81-a5 38 32 7a f9 27 da 3e   c.1.GAz..82z.'.>
Serial number: 0x9826A4
Time stamp: Jun 14 21:59:39 2021 GMT
Accuracy: unspecified
Ordering: yes
Nonce: unspecified
TSA: DirName:/O=Free TSA/OU=TSA/description=This certificate digitally signs documents and time stamp requests made using the freetsa.org online services/CN=www.freetsa.org/emailAddress=busilezas@gmail.com/L=Wuerzburg/C=DE/ST=Bayern
Extensions:

Comment (CleanMachine1) [reply pages/common/openssl-ts.md]: My suggestion uses it, and the file extension specifies it in the command

Comment (0x1080) [reply pages/common/openssl-ts.md]: Ok, I implemented your suggestion.

**Comment by CleanMachine1** [comment]
May I ask, who is https://github.com/258204

**Comment by CleanMachine1** [comment]
Why is he in the participants

**Comment by 0x1080** [comment]
He shouldn't be in the participants. keys got mixed up on the cli git client.

**Comment by 258204** [comment]
Yes, it was a mistake.

**Comment by CleanMachine1** [comment]
Was there a commit where you were the author?

**Comment by 258204** [comment]
I had entered my credentials in git on the computer, so even though Ox1080 made the changes to his branch of the repo, git attributed the commit made in the terminal to me.

**Comment by bl-ue** [comment]
Oh, I see.  No problem — let me see if I can fix it.  Worst possible scenario: we have to close the PR and open a new one (but that's not that bad 🙂).

**Comment by CleanMachine1** [comment]
Im still unsure as to what happened, do you share a computer or similar, SSH keys would probably help you

**Comment by CleanMachine1** [comment]
There is no problem with another contributor.

**Comment by bl-ue** [comment]
That should do it.  I squashed all of the commits that you guys added (for applying @CleanMachine1's suggestions, etc.) into one, and then I changed the author to @0x1080.  All done! 🙂

**Comment by bl-ue** [comment]
(These kinds of Git problems are hard to deal with — experienced Git user here, and they scare me all the time too 😆)

File: pages/common/openssl-ts.md [hunk 7]
```diff
@@ -0,0 +1,20 @@
+# openssl ts
+
+> OpenSSL command to generate and verify timestamps.
+> More information: <https://www.openssl.org/docs/manmaster/man1/openssl-ts.html>.
+
+- Generate SHA-512 timestamp request of `file` (output to `file.tsq`):
```
Comment (bl-ue) [inline pages/common/openssl-ts.md:6]:
```suggestion
- Generate a SHA-512 timestamp request of `file` (output to `file.tsq`):
```

**Comment by 258204** [comment]
Moral of the story is, sharing an ssh key with limited access is not a good way to share a private repo.

File: pages/common/openssl-ts.md [hunk 8]
```diff
@@ -0,0 +1,20 @@
...
+
+`openssl ts -query -data {{path/to/file}}  -sha512  -out {{path/to/file.tsq}}`
+
+- Check the date and metadata from a specified timestamp response file:
```
Comment (bl-ue) [inline pages/common/openssl-ts.md:10]:
```suggestion
- Check the date and metadata of a specific timestamp response file:
```

File: pages/common/openssl-ts.md [hunk 9]
```diff
@@ -0,0 +1,20 @@
...
+
+`openssl ts -reply -in {{path/to/file.tsr}} -text`
+
+- Verify timestamp request `file.tsq` and timestamp response `file.tsr` from server with ssl certificate `cert.pem`:
```
Comment (bl-ue) [inline pages/common/openssl-ts.md:14]:
```suggestion
- Verify timestamp request `file.tsq` and timestamp response `file.tsr` from the server with ssl certificate `cert.pem`:
```

File: pages/common/openssl-ts.md [hunk 10]
```diff
@@ -0,0 +1,20 @@
...
+
+`openssl ts -verify -in {{path/to/file.tsr}} -queryfile {{path/to/file.tsq}} -partial_chain -CAfile {{path/to/cert.pem}}`
+
+- Create time stamp response for request `file.tsq` using key `tsakey.pem` and signing certificate `tsacert.pem` (output to `file.tsr`):
```
Comment (bl-ue) [inline pages/common/openssl-ts.md:18]:
```suggestion
- Create a timestamp response for request `file.tsq` using key `tsakey.pem` and signing certificate `tsacert.pem` (output to `file.tsr`):
```

**Review by bl-ue — COMMENTED** [review]
Looks pretty good to me — I have a few preposition/article comments.

**Comment by CleanMachine1** [comment]
Sharing an SSH key is a terrible idea most likely, it would be like sharing a password.

Why would you share them in the first place, it takes 10 minutes to set up a new one.

**Comment by 258204** [comment]
You can place an ssh public key in your github profile without having the private key, and give it read permission.

**Comment by bl-ue** [comment]
>Moral of the story is, sharing an ssh key with limited access is not a good way to share a private repo.

Yeah, that's not a very good idea 🤣

If you both have your own SSH keys (`ssh-keygen -t rsa -b 4096`, that's all) and upload it to your GitHub account [here](https://github.com/settings/keys), (`cat ~/.ssh/id_rsa.pub | xclip` (or `pbcopy` for macOS or `clip.exe` for Windows)), you can add collaborators to your private repositories, at `https://github.com/<you>/<your_private_or_public_repo>/settings/access` 😉

**Comment by 0x1080** [comment]
The setup is a bit weirder than that, more like a git proxy. Why? Well probably for the same reason some people use ii as an irc client. It's more fun.

**Comment by 258204** [comment]
And yes, in any situation where security is a concern, sharing an ssh key is a terrible idea.

**Comment by 0x1080** [comment]
Anything else needed before approving the commit?

**Comment by CleanMachine1** [comment]
To start, we recently discussed that page PRs should be left for 24 hours so everyone has a chance to reply, however some feel 48 is more acceptable 

I will have my review in around an hour, however I am currently busy and won't review until I have had a good thorough look

File: pages/common/openssl-ts.md [hunk 11]
```diff
@@ -0,0 +1,20 @@
+# openssl ts
+
+> OpenSSL command to generate and verify timestamps.
+> More information: <https://www.openssl.org/docs/manmaster/man1/openssl-ts.html>.
+
+- Generate a SHA-512 timestamp request of `file` (output to `file.tsq`):
```
Comment (CleanMachine1) [inline pages/common/openssl-ts.md:6]:
```suggestion
- Generate a SHA-512 timestamp request of a specific file and output to `file.tsq`:
```

File: pages/common/openssl-ts.md [hunk 12]
```diff
@@ -0,0 +1,20 @@
...
+
+`openssl ts -query -data {{path/to/file}}  -sha512  -out {{path/to/file.tsq}}`
+
+- Check the date and metadata of a specific timestamp response file:
+
+`openssl ts -reply -in {{path/to/file.tsr}} -text`
+
+- Verify timestamp request `file.tsq` and timestamp response `file.tsr` from the server with ssl certificate `cert.pem`:
```
Comment (CleanMachine1) [inline pages/common/openssl-ts.md:14]:
```suggestion
- Verify a timestamp request file and a timestamp response file from the server with ssl certificate file:
```

File: pages/common/openssl-ts.md [hunk 13]
```diff
@@ -0,0 +1,20 @@
...
+
+`openssl ts -verify -in {{path/to/file.tsr}} -queryfile {{path/to/file.tsq}} -partial_chain -CAfile {{path/to/cert.pem}}`
+
+- Create a timestamp response for request `file.tsq` using key `tsakey.pem` and signing certificate `tsacert.pem` (output to `file.tsr`):
```
Comment (CleanMachine1) [inline pages/common/openssl-ts.md:18]:
```suggestion
- Create a timestamp response for request using key and signing certificate and output to `file.tsr`:
```
Leave this one for others

Comment (bl-ue) [reply pages/common/openssl-ts.md:18]: I'd say
```suggestion
- Create a timestamp response for request using key and signing certificate, and output it to `file.tsr`:
```

File: pages/common/openssl-ts.md (hunk 12 above)
Comment (bl-ue) [reply pages/common/openssl-ts.md:14]: nit: `ssl` → `an SSL`

File: pages/common/openssl-ts.md (hunk 13 above)
Comment (0x1080) [reply pages/common/openssl-ts.md:18]: `- Create a timestamp response for request using key and signing certificate, and output the timestamp to `file.tsr`:`

What about?

Comment (bl-ue) [reply pages/common/openssl-ts.md:18]: sgtm 👍🏻

Comment (CleanMachine1) [reply pages/common/openssl-ts.md:18]: the `it` is required however the comma isn't needed

Comment (0x1080) [reply pages/common/openssl-ts.md:18]: > the `it` is required however the comma isn't needed

So
` - Create a timestamp response for request using key and signing certificate and output it to `file.tsr`:`

Comment (CleanMachine1) [reply pages/common/openssl-ts.md:18]: Yes

Comment (bl-ue) [reply pages/common/openssl-ts.md:18]: >the `it` is required however the comma isn't needed

I don't really agree, but @0x1080's [latest](https://github.com/tldr-pages/tldr/pull/6125#discussion_r652132016) looks good to me; let's go with it.

**Review by CleanMachine1 — APPROVED** [review]
:shipit:

**Pull request merged**